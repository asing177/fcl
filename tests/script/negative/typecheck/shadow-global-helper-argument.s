global bool f = True;

@initial
go() {
  foo(f);
  terminate();
}

foo(bool f) {
  !f;
}
