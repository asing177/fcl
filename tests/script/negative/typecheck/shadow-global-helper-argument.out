Error at Line 5, Column 3:
   Expected a statement but got something of type 'bool'

Error at Line 9, Column 10:
   The binding 'f' shadows the global variable
      of the same name, of type 'bool',
      bound at Line 1, Column 17.